lib/puppet/defaults.rb

```diff
@@ -1517,8 +1517,7 @@ def self.default_diffargs
     },
     :lastrunreport =>  {
       :default  => "$statedir/last_run_report.yaml",
-      :type     => :file,
-      :mode     => "0640",
```

Comment:
Well in order to not lose it, it needs a new ::Puppet::Settings::FilePathSetting (or a better name) to be created, to combine the behaviours of type :file and type :path.